### PR TITLE
Refactor geographic gatherering

### PR DESCRIPTION
**Pull request by pnuu**
This PR refactors the code `gatherer.py` uses, deprecates `gatherer.py` in favor of more descriptive `geographic_gatherer.py` and does general clean-up of the code.

### CODE REVIEW

**Comment by pnuu** [comment]
I think this has gotten far enough so that the first round of reviews are called for. I'll be continuing to comb through the code if I can spot something.

**Comment by gerritholl** [comment]
Could you update the documentation too?

**Comment by pnuu** [comment]
Will do! Forgot we had any, but you did add some a while back :grin:

**Comment by gerritholl** [comment]
Will I be able to use the new `geographic_gatherer.py` as a drop-in replacement of `gatherer.py`, or will users need to make any other changes to their configuration?

**Comment by pnuu** [comment]
It's a drop-in replacement. Basically I just renamed the script for clarity, and added a call from the old name to the new one. The major part is the internal restructuring, which shouldn't affect the usage.

File: bin/geographic_gatherer.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Gather granule messages to send them in a bunch."""
```
Comment (mraspaud) [inline bin/geographic_gatherer.py:25]: This should be update to mention 'geographic'

Comment (pnuu) [reply bin/geographic_gatherer.py]: Clarified the docstring.

**Review by mraspaud — APPROVED** [review]
LGTM

File: pytroll_collectors/tests/test_geographic_gatherer.py [hunk 2]
```diff
@@ -81,21 +81,23 @@ def setUp(self):
...
-        self.WatchDogTrigger = self.watchdog_trigger_patcher.start()
-        self.PostTrollTrigger = self.posttroll_trigger_patcher.start()
-        self.get_area_def = self.get_area_def_patcher.start()
-        self.publisher = self.publisher_patcher.start()
-
         self.addCleanup(self.region_collector_patcher.stop)
+
+        self.watchdog_trigger_patcher = patch('pytroll_collectors.geographic_gatherer.WatchDogTrigger')
+        self.WatchDogTrigger = self.watchdog_trigger_patcher.start()
         self.addCleanup(self.watchdog_trigger_patcher.stop)
```
Comment (mraspaud) [inline pytroll_collectors/tests/test_geographic_gatherer.py]:
```suggestion
        self.watchdog_trigger_patcher = patch_cleanup('pytroll_collectors.geographic_gatherer.WatchDogTrigger')
        
        def patch_cleanup(self, item):
            patcher = patch(item)
            patched = patcher.start()
            self.addCleanup(patcher.stop)
            return patched
        
```

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**